src/ImageSharp/Formats/Gif/GifDecoderCore.cs

```diff
@@ -242,6 +295,7 @@ private void ReadLogicalScreenDescriptor()
             {
                 Width = BitConverter.ToInt16(this.buffer, 0),
                 Height = BitConverter.ToInt16(this.buffer, 2),
+                BitsPerPixel = (this.buffer[4] & 0x07) + 1,  // The lowest 3 bits represent the bit depth minus 1
```

Comment:
Was hoping this wouldn't trip you up. Good research. 👍

Reply:
Actually, this was initially done by @xakep139 👍

Reply:
Initially done in https://github.com/SixLabors/ImageSharp/pull/292/commits/f8573e334a4673a31665ee280a10b0e4eb96a15a#diff-7ae85077a97a9237d124f0ddfbe4752c